open_widget_framework/open_widget_framework/views.py

```diff
@@ -1,204 +1,172 @@
         """
-        widget_list = get_object_or_404(WidgetList, id=widget_list_id)
-        # TODO: widget_list.can_access
-        widget_list.clear_list()
-        widget_list.delete()
-        return get_widget_lists(request)
-
+        return make_widget_list_response(self.get_object().get_widgets())
 
-class WidgetView(View):
-    def get(self, request, widget_list_id, widget_id):
+    def create(self, request, *args, **kwargs):
         """
-        API endpoint to get data for a single widget
-
-        Args:
-            widget_id: id of desired Widget
+        API endpoint that creates a new widget list and returns an updated list of widget-list ids
         """
-        widget_list = get_object_or_404(WidgetList, id=widget_list_id)
-        # TODO: widget_list.can_access
+        super().create(request, *args, **kwargs)
+        return self.list(request, *args, **kwargs)
 
-        widget = get_object_or_404(WidgetInstance, id=widget_id)
-
-        # Construct a response dictionary with the widget data as well as the appropriate widget class configuration
-        response = {
-            "widgetData": widget.get_serialized_data(),
-            "widgetClassConfigurations": {
-                widget.widget_class: widget.get_configuration()
-            },
-        }
-        return JsonResponse(response)
-
-    def post(self, request, widget_list_id, widget_id):
+    def destroy(self, request, *args, **kwargs):
         """
-        API endpoint to create a widget instance on a list after validating the data with a serializer
-        class
-
-        Args:
-            widget_list_id: id of WidgetList to add widget to
+        API endpoint that destroys a widget list and all widgets in it and returns an updated list of widget-list ids
         """
-        widget_list = get_object_or_404(WidgetList, id=widget_list_id)
-        # TODO: widget_list.can_access
-        data = loads(request.body.decode())
+        super().destroy(request, *args, **kwargs)
+        return self.list(request, *args, **kwargs)
 
-        # Create a serializer to validate the data
-        serializer = WidgetInstance.get_widget_class_serializer(
-            data.pop("widget_class")
-        )(data=data)
-        if serializer.is_valid():
-            serializer.create_widget(widget_list)
 
-            return WidgetListView.get(request, widget_list_id)
-        else:
-            return JsonResponse({"error": "invalid widget data"}, status=400)
+class WidgetViewSet(ModelViewSet):
+    """
+    WidgetViewSet handles requests at the widget level with the following mapping (as reflected in urls.py):
+        GET -> retrieve
+        POST -> create
+        DELETE -> destroy
+        PUT -> update
+        PATCH -> partial_update
+    """
+    serializer_class = WidgetSerializer
 
-    def delete(self, request, widget_list_id, widget_id):
+    def check_widget_list_edit_permissions(self):
```

Comment:
If you make my suggested change to add a second setting for this viewset then I don't think you need this method anymore.

Reply:
@rhysyngsun Merging in this PR for now, but if this is unresolved, we can address it in a later PR